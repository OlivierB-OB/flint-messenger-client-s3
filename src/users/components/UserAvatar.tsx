import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import statusBadgeFactory from './statusBadgeFactory';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import { IUserInfo } from '../types';

const anchorOrigin = {
  vertical: 'bottom',
  horizontal: 'right',
};

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    secondary: {
      color: theme.palette.getContrastText(theme.palette.secondary.dark),
      backgroundColor: theme.palette.secondary.dark,
    },
  }),
);

export interface IUserAvatarProps {
  info?: IUserInfo;
}

// FIXME support profile picture

export function UserAvatar({ info }: IUserAvatarProps) {
  const { lastName, firstName, status } = info || { firstName: '?', lastName: '?', status: 'offline' };
  const initials = `${firstName[0]}${lastName[0]}`;
  const StatusBadge = statusBadgeFactory(status);
  const classes = useStyles();
  return (
    <StatusBadge variant="dot" overlap="circle" anchorOrigin={anchorOrigin}>
      <Avatar className={classes.secondary}>{initials}</Avatar>
    </StatusBadge>
  );
}
