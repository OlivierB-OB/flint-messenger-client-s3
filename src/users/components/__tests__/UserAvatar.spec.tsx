import React from 'react';
import { shallow } from 'enzyme';
import { UserAvatar } from '../UserAvatar';

describe('UserAvatar', () => {
  it('should display the user initial', async () => {
    const info: any = {
      firstName: 'foo',
      lastName: 'bar',
      status: 'offline',
    };
    const component = shallow(<UserAvatar info={info} />);
    expect(component).toMatchSnapshot();
  });

  it('should support undefined user information', async () => {
    const component = shallow(<UserAvatar info={undefined} />);
    expect(component).toMatchSnapshot();
  });
});
