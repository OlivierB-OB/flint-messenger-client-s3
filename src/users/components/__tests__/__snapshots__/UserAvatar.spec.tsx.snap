// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar should display the user initial 1`] = `
<WithStyles(WithStyles(ForwardRef(Badge)))
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "bottom",
    }
  }
  overlap="circle"
  variant="dot"
>
  <WithStyles(ForwardRef(Avatar))
    className="makeStyles-secondary-1"
  >
    fb
  </WithStyles(ForwardRef(Avatar))>
</WithStyles(WithStyles(ForwardRef(Badge)))>
`;

exports[`UserAvatar should support undefined user information 1`] = `
<WithStyles(WithStyles(ForwardRef(Badge)))
  anchorOrigin={
    Object {
      "horizontal": "right",
      "vertical": "bottom",
    }
  }
  overlap="circle"
  variant="dot"
>
  <WithStyles(ForwardRef(Avatar))
    className="makeStyles-secondary-1"
  >
    ??
  </WithStyles(ForwardRef(Avatar))>
</WithStyles(WithStyles(ForwardRef(Badge)))>
`;
